src/storage/src/hummock/compactor/iterator.rs

```diff
@@ -291,9 +309,15 @@ impl ConcatSstableIterator {
                 &self.stats,
             );
             sstable_iter.seek(seek_key).await?;
-
-            self.sstable_iter = Some(sstable_iter);
             self.cur_idx = idx;
+
```

Comment:
Should we also consider `sstable_info.get_table_ids()` when searching for the `start_index` in L268? We may save some unneccesary I/Os.